opentracing/tracer.py

```diff
@@ -56,60 +71,118 @@ def start_span(self,
         `format` is unknown or disallowed.
 
         Implementations may raise opentracing.InvalidCarrierException,
-        opentracing.TraceCorruptedException, or implementation-specific errors
-        if there are problems with `carrier`.
+        opentracing.SpanContextCorruptedException, or implementation-specific
+        errors if there are problems with `carrier`.
 
-        Upon success, the returned Span instance is already started.
-
-        :param operation_name: the operation name for the returned Span (which
-            can be set later via Span.set_operation_name())
         :param format: a python object instance that represents a given
             carrier format. `format` may be of any type, and `format` equality
             is defined by python `==` equality.
-        :param carrier: the format-specific carrier object to join with
+        :param carrier: the format-specific carrier object to extract from
 
-        :return: a Span instance joined to the trace state in `carrier` or None
-            if no such trace state could be found.
+        :return: a SpanContext instance extracted from `carrier` or None if no
+            such span context could be found.
         """
-        return self._noop_span
+        return self._noop_span_context
 
-    def flush(self):
-        """Flushes any trace data that may be buffered in memory, presumably
-        out of the process.
 
-        :return: Returns a :py:class:futures.Future
-        """
-        fut = Future()
-        fut.set_result(True)
-        return fut
+class ReferenceType:
+    """A namespace for OpenTracing reference types.
+
+    See http://opentracing.io/spec for more detail about references, reference
+    types, and CHILD_OF and FOLLOWS_FROM in particular.
+    """
+    CHILD_OF = 'child_of'
+    FOLLOWS_FROM = 'follows_from'
+
+
+# We inherit from namedtuple (rather than assign) so that we can specify a
+# standard docstring.
+class Reference(namedtuple('Reference', ['type', 'span_context'])):
+    """A Reference pairs a reference type with a SpanContext referee.
+
+    References are used by Tracer.start_span() to describe the relationships
+    between Spans.
+    """
+    pass
+
+
+def ChildOf(span_context):
+    """ChildOf is a helper that creates CHILD_OF References.
+
+    :param span_context: the (causal parent) SpanContext to reference
+
+    :rtype: Reference
+    :return: A Reference suitable for Tracer.start_span(..., references=...)
+    """
+    return Reference(ReferenceType.CHILD_OF, span_context)
+
+
+def FollowsFrom(span_context):
+    """FollowsFrom is a helper that creates FOLLOWS_FROM References.
+
+    :param span_context: the (causal parent) SpanContext to reference
+
+    :rtype: Reference
+    :return: A Reference suitable for Tracer.start_span(..., references=...)
+    """
+    return Reference(ReferenceType.FOLLOWS_FROM, span_context)
+
+
+def start_child_span(parent_span, operation_name, tags=None, start_time=None):
```

Comment:
since we do not have functional options, it makes sense to  handle `RPCServerOption` via a similar helper method, e.g. `start_rpc_server_span`